core/lib/zksync_core/src/api_server/tx_sender/mod.rs

```diff
@@ -906,8 +908,24 @@ impl<G: L1GasPriceProvider> TxSender<G> {
             H256::zero()
         };
 
-        let seal_data = SealData::for_transaction(transaction, tx_metrics);
-        if let Some(reason) = ConditionalSealer::find_unexecutable_reason(sk_config, &seal_data) {
+        let protocol_version = self
+            .0
+            .replica_connection_pool
+            .access_storage_tagged("api")
+            .await
+            .unwrap()
+            .blocks_dal()
+            .get_last_sealed_miniblock_header()
+            .await
+            .unwrap()
+            .unwrap_or_default()
+            .protocol_version
+            .unwrap_or_default();
```

Comment:
nit, you can unwrap in both places safely:
1. Unwrapping header is safe because genesis header must be present before running API.
2. Protocol versions are not set only for some historical blocks, so it's safe to unwrap `protocol_version` of **latest** block

Then you won't need `MiniblockHeader` to implement `Default`.